### ISSUE TITLE
Fix S3963: Should not raise issue when the static constructor does not assign any fields

### ISSUE THREAD

**Issue by dee-see** [SonarSource/sonar-csharp#366]
### Description

Rule S3963, as far as I understand by reading the code, raises an error as soon as there is a static constructor. That picks up much more cases than the "Initialize all 'static fields' inline and remove the 'static constructor'" description would suggest.

### Repro steps

```
    class Program
    {
        static Program() 
        {
            System.Console.WriteLine("Not assigning a field");
        }
    }
```

Real life examples instead of a `WriteLine`: initialization of GUI component, static initialisation of a vendor's license, basically all sorts of `void` method calls with side effects.

### Expected behavior

No issue found

### Actual behavior

S3963 is raised

### Known workarounds

Turning off the rule or flagging as false positive

### Related information

SonarQube 6.3.1
SonarCSharp 5.10.1 (build 1411)
MSBuild 14

**Comment by Evangelink** [comment]
[RSPEC-3963](https://jira.sonarsource.com/browse/RSPEC-3963)

**Comment by levindv** [comment]
The same issue in VS2022 for SonarLint 6.9.0.54300.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @levindv, 

Please create a new issue with the reproducer and a clear description of the problem. This one has been closed 5 years ago.
